lib/xlsxtream/worksheet.rb

```diff
@@ -4,12 +4,8 @@
 
 module Xlsxtream
   class Worksheet
-    attr_reader :id, :name
-
-    def initialize(io, id, name, options = {})
+    def initialize(io, options = {})
```

Comment:
@felixbuenemann You are probably already aware, but Ruby will remove the implicit conversion between last-argument-hash and keyword arguments in Ruby 3.0.

https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

Right now a mix of the syntaxes are used:

```ruby
ws = Worksheet.new(StringIO.new, id: 1)

# …

worksheet = Worksheet.new(@io, :id => sheet_id, :name => name, …)
```

Reply:
I'm not using keyword arguments at all at the moment, so this isn't really an issue.

However I'll probably just convert the options hashes to keyword arguments in the future.

I think the first versions of xlsxtream supported very old ruby versions, so I couldn't use kwargs back then.

I don't think there are any plans to prevent passing a hash without braces as the last argument, if the method doesn't take any keyword arguments. At least Ruby 2.7 doesn't emit warnings for that.

Reply:
Alright, makes sense! Again, thanks for working with @kukicola and me on this! 💯